content/articles/regex-implementation-with-java/index.md

```diff
@@ -0,0 +1,394 @@
+
+#### ASSERTIONS
+
+Assertions are used to specify boundaries within which a match should occur. They are also referred to as regex anchors.
+
+
+
+| Character | Meaning|
+| ----------|--------
+|  `^`      | This ensures that the match starts at the beginning of the string or line. However, when this is used at the start of character groups or ranges, it ensures that every other character except the specified characters in the group is a match.|
+| `$`       | This ensures that the match occurs at the end of the string or before a newline character.|
+| `\b`      | This denotes a word boundary and matches the occurrence of characters not preceded or followed by any word character. For example, string “anna” matches the pattern `\\b\\w+` because any other word character does not precede the string. Still, it is not a match for `a\\b` because the first character ‘a’ is immediately preceded by other word characters.|
+| `\B`      | This matches a non-word boundary, given a position where the previous and next characters are either both words or both non-words.|
+
+#### CHARACTER ESCAPE (`\`)
+
+The backslash `\` is used to obtain the literal meaning or value, usually a quantifier or any other special character. For example, to obtain the literal backslash character, it is required to be escaped `\\`.
+
+It is also used to indicate that the character that follows it is special, as described in the character classes section of this article.
+
+### JAVA PATTERN AND MATCHER APIs
+
+The Java class Pattern represents a compiled regular expression. The regex pattern is created using the `Pattern.compile()` method. This is an overloaded method whose first or only parameter (depending on its method is invoked) is a String. The String argument contains the regex pattern to be matched. 
+
+The Java class Matcher performs match operations on Strings or character sequences by interpreting a compiled regex pattern.
+
+```java
+
+
+    public static void main(String[] args){
+        String statements = """
+                Paul's wedding was 09/15/2002
+                Today's date is 07/16/2015
+                Priscilla graduated on 10/25/20
+                """;
+
+        Pattern compiledRegex = Pattern.compile("P.*\\d{1,2}/\\d{1,2}/\\d{1,2}");
+        Matcher regexMatcher = compiledRegex.matcher(statements);
+
+        while(regexMatcher.find())
+            System.out.println(regexMatcher.group());
+
+//          output: Paul's wedding was 09/15/20
+//                  Priscilla graduated on 10/25/20  
+
+
+    }
+```
+
+
+In the code above, the Java class `Pattern` compiles the regex pattern and stores the value in the variable `compiledRegex` - an instance of the `Pattern` class which then creates the Matcher instance, `regexMatcher`, which also holds the returned value of the compiled regex pattern matching the string literal with variable name `statements`.
```

Comment:
break down this sentence

Reply:
@paulodhiambo all suggested modifications have been implemented and committed. Thanks